doipclient/client.py

```diff
@@ -288,6 +296,65 @@ def await_vehicle_announcement(
             if result and type(result) == VehicleIdentificationResponse:
                 return addr, result
 
+    @classmethod
+    def request_vehicle_identification(cls, ipv6=False, source_interface=None, ecu_ip_address='255.255.255.255', protocol_version=0x02, eid=None, vin=None):
+        """Requests a VehicleIdentificationResponse from the ECU, either with a specified VIN, EIN,
+        or nothing.
+
+        :param ipv6: Bool forcing IPV6 socket instead of IPV4 socket
+        :type ipv6: bool, optional
+        :param source_interface: Interface name (like "eth0") to bind to for use with IPv6. Defaults to None which
+            will use the default interface (which may not be the one connected to the ECU). Does nothing for IPv4,
+            which will bind to all interfaces using INADDR_ANY.
+        :type source_interface: str, optional
+        :param ecu_ip_address: This is the IP address of the target ECU for unicast. Defaults to broadcast if
+        the address is not known. 
+        :type ecu_ip_address: str, optional
+        :param protocol_version: The DoIP protocol version to use for communication. Represents the version of the ISO 13400
+            specification to follow. 0x02 (2012) is probably correct for most ECU's at the time of writing, though technically
+            this implementation is against 0x03 (2019).
+        :type protocol_version: int, optional
+        :param eid: EID of the Vehicle
+        :type eid: bytes, optional
+        :param vin: VIN of the Vehicle
+        :type vin: str, optional
+        :return: The vehicle identification response message
+        :rtype: VehicleIdentificationResponse
+        """
+
+        # UDP_TEST_EQUIPMENT_REQUEST is dynamically assigned using udp_port=0
+        sock = cls.create_udp_socket(udp_port=0, ipv6=ipv6, timeout=A_DOIP_CTRL, source_interface=source_interface)
+
+        if eid:
+            message = VehicleIdentificationRequestWithEID(eid)
+        elif vin:
+            message = VehicleIdentificationRequestWithVIN(vin)
+        else:
+            message = VehicleIdentificationRequest()
+
+        payload_data = message.pack()
+        payload_type = payload_message_to_type[type(message)]
+
+        data_bytes = struct.pack(
```

Comment:
some code duplication from `send_doip` https://github.com/jacobschaer/python-doipclient/blob/1a1403eb7bb741139a5c32d8c0f9ff593fea2f02/doipclient/client.py#L415-L429

I can put it in a staticfunction if you want @jacobschaer ?

Reply:
I see what you mean - that is somewhat inconvenient. Probably should move that out to a new function like you suggested, though leave the logger message out. I think it's ok for this new method to have its own unique logger message


```
@staticmethod
def pack_doip(protocol_version, payload_type, payload_data):
     data_bytes = struct.pack( 
	     "!BBHL", 
	     self._protocol_version, 
	     0xFF ^ self._protocol_version, 
	     payload_type, 
	     len(payload_data), 
	 ) 
	data_bytes += payload_data
	return data_bytes
```

Then in send_doip()... call it with

```
self.pack_doip(self._protocol_version, payload_type, payload_data)
logger.debug( 
     "Sending DoIP Message: Type: 0x{:X}, Payload Size: {}, Payload: {}".format( 
         payload_type, 
         len(payload_data), 
         " ".join(f"{byte:02X}" for byte in payload_data), 
     ) 
)
```

And in your new function maybe some new log message like "Sending DoIP Vehicle Information Request..."

Reply:
Done!